buildspec.pypi.yml

```diff
@@ -0,0 +1,36 @@
+version: 0.2
+
+env:
+  shell: bash
+  secrets-manager:
+    PYPI_TOKEN: "pypi:token"
+
+phases:
+  install:
+    commands:
+      - echo Setting local Python versions
+      - pyenv versions | awk 'match($0, /\d\.\d+\.\d+/) { print substr($0, RSTART, RLENGTH) }' > .python-version
+      - echo Installing dependencies
+      - pip install poetry
+      - poetry install
+
+  build:
+    commands:
+      - echo Running check-cloudformation
+      - poetry run ./copilot_helper.py check-cloudformation
+      - echo Running tests
+      - poetry run tox
+      - |
+        if ! python utils/check_pypi.py
```

Comment:
It feels like a lot of this could be replaced with the following.

```shell
REMOTE_VERSION="$(curl -s "https://pypi.org/pypi/dbt-copilot-tools/json" | jq -rc '.info.version')"
LOCAL_VERSION="$(cat pyproject.toml | grep "^version" | awk '{ print $3; }' | jq -rc)"

if [[ $LOCAL_VERSION > $REMOTE_VERSION ]]; then
  echo "updating"
else
  echo "not updating"
fi
```

Though happy with a python script if we feel that is more maintainable and tests are added to it.

Reply:
I prefer the python as it is much more readable than awk/grep/jq chains. I'll add some tests.

Reply:
Didn't realise there was a json endpoint in PyPI though, so I'll change to that from the rss feed, as searching the XML is fairly horrible.